src/ensembl/src/content/app/browser/browser-region-editor/BrowserRegionEditor.tsx

```diff
@@ -214,22 +210,18 @@ export const BrowserRegionEditor = (props: BrowserRegionEditorProps) => {
             errorMessages.locationEndError
           );
         } else {
-          updateAllInputs();
-          setIsFormSubmitted(false);
-          updateErrorMessages(null, null);
-          props.toggleRegionEditorActive(false);
-
+          resetForm();
           updateBrowser();
         }
       }
     }
-  }, [props.regionValidationInfo]);
+  }, [props.validationInfo]);
```

Comment:
Doesn’t it essentially mean "every time the component receives the new props"? ValidationInfo is an object, and props comparison for objects is by reference (whether it’s the same object rather than whether it has the same values), and I suspect every time the component updates due to props change, `props.validationInfo` will be different. Does this effect behave differently with `props.validationInfo` than without any dependencies array? If yes, then something is wrong with the dependency.

Reply:
The intention is to run this whenever `props.validationInfo` gets updated as that would mean the response has been returned. The validation happens only whenever a user clicks on the `submit` button, hence not many requests to validate. So I thought premature optimisation for this is not necessary.

Reply:
This is not a matter of premature optimisation.
This is (if I understand correctly how it will work; I may be wrong) a misleading piece of code.
It suggests, like you say, "run this whenever props.validationInfo gets updated". But what it likely is actually doing (again, unless React is somehow smarter than I take it to be) is "run this whenever props get updated". Because every time the props are updated, the validationInfo object will be different.
If that's what it's doing (and if it's ok), then removing the list of dependencies will help avoid the misunderstanding.

Alternatively, there are [two custom hooks](https://github.com/Ensembl/ensembl-client/commit/bf65e4f0cc3fdcf796bbb3827be236c602a351de) that I wrote (well, copied mostly) a while back, but never had a chance to actually use. One of them, `useDeepCompareEffect`, does deep comparison of props. If you want to compare the validationInfo prop, we should use that.

I am still not sure this useEffect is doing the proper thing though. It:
- updates the state to show error messages (why do we keep them in the component's local state? why don't we show them directly as they come from props?
- updates the browser position (why don't we do it from inside the action that performs validation)
- clears the form (why don't we pass a callback to the action that does the validation, so that it clears the form on success?)

Reply:
As we discussed offline a couple of times about this, the `useEffect` is not used anymore and the `validateRegion` function has been moved into the browser helper.